### PR TITLE
Introduce persistence milestoning Physical plan and Relational Core

### PR DESCRIPTION
**Pull request by prasar-ashutosh**
Introduce persistence milestoning Physical plan and Relational Core

#### What type of PR is this?

Feature.

#### What does this PR do / why is it needed ?

We are introducing a common library to handle batch ingestion logic with an initial focus on batch milestoning into relational databases. This library consolidates the implementation across multiple backend persistence implementations. It also provides an in-memory implementation that will be of use for fast testing.

This PR introduces the two component of the solution above: physical plan and relational core

#### Which issue(s) this PR fixes:
None. But this is under the series of PRs addressing https://github.com/finos/legend-engine/issues/920

Fixes #

#### Does this PR introduce a user-facing change?
No.

### CODE REVIEW

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        </dependency>
+
+        <!-- TEST -->
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
```
Comment (rafaelbey) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/pom.xml:62]: Need test scope

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        <!-- LOGGING -->
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+        </dependency>
+
+        <!-- TEST -->
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
```
Comment (rafaelbey) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/pom.xml:58]: Need test scope

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/pom.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        <!-- MISC -->
+        <dependency>
+            <groupId>commons-codec</groupId>
+            <artifactId>commons-codec</artifactId>
+        </dependency>
+
+        <!-- TEST -->
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
```
Comment (rafaelbey) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/pom.xml:63]: Need test scope

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/pom.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
```
Comment (rafaelbey) [inline legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/pom.xml:67]: Need test scope

File: legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/pom.xml (hunk 1 above)
Comment (davidsunglee) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/pom.xml:62]: Scope should be inherited from the parent pom dependency management section

Comment (davidsunglee) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/pom.xml:62]: david@Colossus:~/Code/IDEA/legend-engine-fork/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan$ mvn dependency:tree
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] --< org.finos.legend.engine:legend-engine-xt-persistence-component-physical-plan >--
[INFO] Building Legend Engine - XT - Persistence - Component - Physical Plan 3.15.4-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from ossrh: https://s01.oss.sonatype.org/content/repositories/snapshots/org/finos/legend/engine/legend-engine-xt-persistence-component-logical-plan/3.15.4-SNAPSHOT/maven-metadata.xml
Downloading from ossrh-releases: https://s01.oss.sonatype.org/content/repositories/releases/org/finos/legend/engine/legend-engine-xt-persistence-component-logical-plan/3.15.4-SNAPSHOT/maven-metadata.xml
Downloading from ossrh-releases: https://s01.oss.sonatype.org/content/repositories/releases/org/finos/legend/engine/legend-engine-xt-persistence-component/3.15.4-SNAPSHOT/maven-metadata.xml
Downloading from ossrh: https://s01.oss.sonatype.org/content/repositories/snapshots/org/finos/legend/engine/legend-engine-xt-persistence-component/3.15.4-SNAPSHOT/maven-metadata.xml
Downloading from ossrh-releases: https://s01.oss.sonatype.org/content/repositories/releases/org/finos/legend/engine/legend-engine/3.15.4-SNAPSHOT/maven-metadata.xml
Downloading from ossrh: https://s01.oss.sonatype.org/content/repositories/snapshots/org/finos/legend/engine/legend-engine/3.15.4-SNAPSHOT/maven-metadata.xml
Downloaded from ossrh: https://s01.oss.sonatype.org/content/repositories/snapshots/org/finos/legend/engine/legend-engine/3.15.4-SNAPSHOT/maven-metadata.xml (615 B at 8.7 kB/s)
[INFO] 
[INFO] --- maven-dependency-plugin:2.10:tree (default-cli) @ legend-engine-xt-persistence-component-physical-plan ---
[INFO] org.finos.legend.engine:legend-engine-xt-persistence-component-physical-plan:jar:3.15.4-SNAPSHOT
[INFO] +- org.finos.legend.engine:legend-engine-xt-persistence-component-logical-plan:jar:3.15.4-SNAPSHOT:compile
[INFO] +- org.immutables:value:jar:2.8.2:provided
[INFO] +- com.fasterxml.jackson.core:jackson-core:jar:2.10.5:compile
[INFO] +- com.fasterxml.jackson.core:jackson-databind:jar:2.10.5.1:compile
[INFO] |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.10.5:compile
[INFO] +- commons-codec:commons-codec:jar:1.15:compile
[INFO] +- org.junit.jupiter:junit-jupiter-api:jar:5.4.0:test
[INFO] |  +- org.apiguardian:apiguardian-api:jar:1.0.0:test
[INFO] |  +- org.opentest4j:opentest4j:jar:1.1.1:test
[INFO] |  \- org.junit.platform:junit-platform-commons:jar:1.4.0:test
[INFO] \- org.junit.jupiter:junit-jupiter-engine:jar:5.4.0:test
[INFO]    \- org.junit.platform:junit-platform-engine:jar:1.4.0:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.469 s
[INFO] Finished at: 2022-09-14T11:15:48-04:00
[INFO] ------------------------------------------------------------------------

Comment (davidsunglee) [reply legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-relational-core/pom.xml:62]: Verified via CLI just to be sure

**Review by rafaelbey — APPROVED** [review]

**Pull request merged**